runtime/src/main/java/org/corfudb/util/serializer/ISerializer.java

```diff
@@ -54,6 +55,14 @@ protected Kryo initialValue() {
      */
     Object deserialize(ByteBuf b, CorfuRuntime rt);
 
+    /**
+     * Deserialize an object from a given byte buffer.
+     *
+     * @param b The bytebuf to deserialize.
+     * @return The deserialized object.
+     */
+    <T> T deserializeTyped(ByteBuf b, CorfuRuntime rt);
```

Comment:
Seems like there is no usage of this method. Why do we need it then?

Reply:
Hm, the part that uses deserializeTyped is in #3814, when I separated the main pr to this one and to 3814 I couldn't bring those changes from 3814 in this pr. This method will be used but in the next pr

Reply:
Why do you need it? Why wouldn't `Object deserialize(ByteBuf b, CorfuRuntime rt);` suffice? Seems like you are trying to get rid of `Unchecked cast: 'java.lang.Object' to 'K' `messages, however, you are just effectively hiding it.

Reply:
Also, you are forcing all other interfaces to implement it.